core/src/main/java/com/netflix/conductor/core/execution/mapper/SwitchTaskMapper.java

```diff
@@ -96,8 +96,7 @@ public List<TaskModel> getMappedTasks(TaskMapperContext taskMapperContext) {
         switchTask.getInputData().put("case", evalResult);
         switchTask.getOutputData().put("evaluationResult", Collections.singletonList(evalResult));
         switchTask.setStartTime(System.currentTimeMillis());
-        switchTask.setStatus(TaskModel.Status.COMPLETED);
-        switchTask.setExecuted(true);
+        switchTask.setStatus(TaskModel.Status.IN_PROGRESS);
```

Comment:
The switch task is completed here so the status should be `COMPLETED` .Only executed flag has to be removed.

Reply:
It will be marked as completed in the executor